Source/Scene/Expression.js

```diff
@@ -881,5 +925,182 @@ define([
+                // Left can be an array if the type is LITERAL_COLOR
+                var length = this._left.length;
+                left = new Array(length);
+                for (var i = 0; i < length; ++i) {
+                    var shader = this._left[i].getShaderExpression(variablePrefix, info);
+                    if (!defined(shader)) {
+                        // If the left side is not valid shader code, then the expression is not valid
+                        return undefined;
+                    }
+                    left[i] = shader;
+                }
+            } else {
+                left = this._left.getShaderExpression(variablePrefix, info);
+                if (!defined(left)) {
+                    // If the left side is not valid shader code, then the expression is not valid
+                    return undefined;
+                }
+            }
+        }
+
+        if (defined(this._right)) {
+            right = this._right.getShaderExpression(variablePrefix, info);
+            if (!defined(right)) {
+                // If the right side is not valid shader code, then the expression is not valid
+                return undefined;
+            }
+        }
+
+        if (defined(this._test)) {
+            test = this._test.getShaderExpression(variablePrefix, info);
+            if (!defined(test)) {
+                // If the test is not valid shader code, then the expression is not valid
+                return undefined;
+            }
+        }
+
+        switch (type) {
+            case ExpressionNodeType.VARIABLE:
+                return variablePrefix + value;
+            case ExpressionNodeType.UNARY:
+                // Supported types: +, -, !, Boolean, Number
+                if ((value === 'isNan') || (value === 'isFinite') || (value === 'String')) {
+                    return undefined;
+                } else if (value === 'Boolean') {
+                    return 'bool(' + left + ')';
+                } else if (value === 'Number') {
+                    return 'float(' + left + ')';
+                }
+                return value + left;
+            case ExpressionNodeType.BINARY:
+                // Supported types: ||, &&, ===, !==, <, >, <=, >=, +, -, *, /, %
+                if (value === '%') {
+                    return 'mod(' + left + ', ' + right + ')';
+                } else if (value === '===') {
+                    return '(' + left + ' == ' + right + ')';
+                } else if (value === '!==') {
+                    return '(' + left + ' != ' + right + ')';
+                }
+                return '(' + left + ' ' + value + ' ' + right + ')';
+            case ExpressionNodeType.CONDITIONAL:
+                return '(' + test + ' ? ' + left + ' : ' + right + ')';
+            case ExpressionNodeType.MEMBER:
+                // This is intended for accessing the components of vec2, vec3, and vec4 properties. String members aren't supported.
```

Comment:
@hobu are strings common per-point metadata?

@lilleyse if so, I think we need to put the characters into textures to support this.  If not, it may be reasonable for the spec to disallow strings in styles for point clouds.